src/formatDate.js

```diff
@@ -9,6 +9,72 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const separatorFrom = fromFormat.pop();
+  const separatorTo = toFormat.pop();
+  const dateParts = date.split(separatorFrom);
+  const formattedDate = [];
+
+  let oldIndexDay = 0;
+  let oldIndexMonth = 0;
+  let oldIndexYear = 0;
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    if (fromFormat[i][0] === 'Y') {
+      oldIndexYear = i;
+    }
+
+    if (fromFormat[i][0] === 'M') {
+      oldIndexMonth = i;
+    }
+
+    if (fromFormat[i][0] === 'D') {
+      oldIndexDay = i;
+    }
+  }
+
+  let newIndexDay = 0;
+  let newIndexMonth = 0;
+  let newIndexYear = 0;
+
+  for (let i = 0; i < toFormat.length; i++) {
+    if (toFormat[i][0] === 'Y') {
+      newIndexYear = i;
+    }
+
+    if (toFormat[i][0] === 'M') {
+      newIndexMonth = i;
+    }
+
+    if (toFormat[i][0] === 'D') {
+      newIndexDay = i;
+    }
```

Comment:
Similarly to the previous comment, you can also optimize the loop for finding the new indexes by combining it with the previous loop. This will reduce the number of iterations and make the code more concise.